src/config/createConfig.ts

```diff
@@ -140,7 +140,9 @@ export const createConfig = (userConfig: UserConfig): InternalConfig => {
       loadPath: `${clientLocalePath}/${localeStructure}.${localeExtension}`,
     }
 
-    combinedConfig.ns = [defaultNS]
+    if (!combinedConfig.ns) {
```

Comment:
The `ns` option can either be a string or an array, so let's explicitly check:

```suggestion
    if (!typeof combinedConfig.ns === 'string' || Array.isArray(combinedConfig.ns)) {
```

Reply:
Done 👍